tasks/src/main/java/ru/sberbank/school/task02/calculator/CompareQuotesBenificiary.java

```diff
@@ -0,0 +1,46 @@
+package ru.sberbank.school.task02.calculator;
+
+import ru.sberbank.school.task02.util.Beneficiary;
+import ru.sberbank.school.task02.util.ClientOperation;
+import ru.sberbank.school.task02.util.Quote;
+
+import java.math.BigDecimal;
+import java.util.Comparator;
+
+public class CompareQuotesBenificiary implements Comparator<Quote> {
+    private Beneficiary beneficiary;
+    private ClientOperation operation;
+
+    public CompareQuotesBenificiary (ClientOperation operation, Beneficiary beneficiary) {
+        this.beneficiary = beneficiary;
+        this.operation = operation;
+    }
+
+    @Override
+    public int compare(Quote quote1, Quote quote2) {
+        BigDecimal quoteVolume1 =  quote1.getVolume().getVolume();
+        BigDecimal quoteVolume2 =  quote2.getVolume().getVolume();
+        if (quote1.getVolume().isInfinity()) {
+            return 1;
+        }
+        if (quote2.getVolume().isInfinity()) {
+            return -1;
+        }
+        if (quoteVolume1.compareTo(quoteVolume2) == 0) {
+            return compareEqualQuotes(quote1, quote2);
+        }
+        return quoteVolume1.compareTo(quoteVolume2);
+    }
+
+    private int compareEqualQuotes(Quote quote1, Quote quote2) {
+        if ((beneficiary == Beneficiary.BANK && operation == ClientOperation.BUY) ||
```

Comment:
&#39;||&#39; should be on a new line.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-1209055422>*